pkg/kudoctl/util/resources/package.go

```diff
@@ -0,0 +1,49 @@
+package resources
+
+import (
+	"os"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/http"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/finder"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+)
+
+// GetPackageResources tries to look for package files resolving the operator name to:
+// - a local tgz file
+// - a local directory
+// - a url to a tgz
+// - an operator name in the remote repository
+// in that order. Should there exist a local folder e.g. `cassandra` it will take precedence
+// over the remote repository package with the same name.
+func GetPackageResources(operatorName string, version string, repository repo.Repository) (*packages.PackageCRDs, error) {
```

Comment:
Why is this method not part of the `/packages` pkg? It seems like a better place for it since it is dealing with packaged files (either locally or over the wire). 

Also, naming:
1. `Get`-prefix is also considered an anti-pattern `GetPackageResources() -> packages.Resources()`
2. This method should also return `*packages.Resources` instead of `PackageCRDs`

Reply:
:+1:

Reply:
Unfortunately, this can't be moved into `packages`, because this will result in import cycles due to circular package references. This is due to tight coupling in this as well as other packages. Further refactorings are necessary to clean this up and are out of scope of this PR. I'll move this to the `kudo` package for now.